skill-tree /src/main/java/com/RDS/skilltree/Endorsement/EndorsementController.java

```diff
@@ -0,0 +1,44 @@
+package com.RDS.skilltree.Endorsement;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Map;
+import java.util.UUID;
+
+import static com.RDS.skilltree.Endorsement.JsonApiResponseConverter.convertToHashMap;
+
+@RestController
+@RequestMapping("/v1/endorsements")
+@Slf4j
+public class EndorsementController {
+    private final EndorsementService endorsementService;
+
+    public EndorsementController(EndorsementService endorsementService) {
+        this.endorsementService = endorsementService;
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<?> getEndorsementById(@PathVariable(value = "id", required = true) String id){
+        try {
+            UUID uuid = UUID.fromString(id);
```

Comment:
The incoming request will have UUID in string format. hence we need to convert it into UUID. without this conversion it was throwing a error expected UUID received string.

Reply:
Can you have the dataype of `id` as UUID then you are accepting the input as UUID, not string, you don't require to convert to UUID again.

Reply:
If I modify the URL parameter to be of type UUID and an invalid UUID is passed as a parameter, I encounter difficulty throwing an exception from within the controller because the control does not reach inside the controller function.

we will see this in the terminal 

`2023-12-02T15:06:04.649+05:30  WARN 93592 --- [nio-8080-exec-1] .w.s.m.s.DefaultHandlerExceptionResolver : Resolved [org.springframework.web.method.annotation.MethodArgumentTypeMismatchException: Failed to convert value of type 'java.lang.String' to required type 'java.util.UUID'; UUID string too large]`

Reply:
If you see the parameter is inherently of type String and Spring boot is trying to convert it into UUID @vikhyat187 and the request is terminated before reaching the controller file.

Reply:
Yes I think, this is the validation which Spring does on UUID and return error incase the UUID is incorrect @heyrandhir